Ramos_HwAssignments/Ramos_ClassAssignment_BarcodeScanner/BarcodeScanner.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useEffect } from 'react';
+import { Text, View, StyleSheet, Button } from 'react-native';
+import { BarCodeScanner } from 'expo-barcode-scanner';
+import { NavigationContainer, useNavigation, useRoute, RouteProp, StackActions } from '@react-navigation/native';
+import { StackNavigationProp } from '@react-navigation/stack';
+import { StackParamList } from './App';
+import { useFetch } from './Functions/useFetch';
+
+type ScannerNavigationProp = StackNavigationProp<ScannerStackParamList, 'BarcodeScanner'>;
+
+export const BarcodeScanner = () => {
+  const navigation = useNavigation<ScannerNavigationProp>();
+
+  const [hasPermission, setHasPermission] = useState(null);
+  const [scanned, setScanned] = useState(false);
+
+  const defaultURL = {data: "https://rn-products-1d8a6-default-rtdb.firebaseio.com/products/1.json"};
+  const [productURL, setProductURL] = useState("");
+  const {data, loading, error} = useFetch(productURL);
+
+  useEffect(() => {
+    const getBarCodeScannerPermissions = async () => {
+      const { status } = await BarCodeScanner.requestPermissionsAsync();
+      setHasPermission(status === 'granted');
+    };
+
+    getBarCodeScannerPermissions();
+  }, []);
+
+  useEffect(() => {
+    if (data != null)
+      navigation.navigate('ProductDetail', {
+        data: data,
+    });
+  }, [data]);
```

Comment:
If we kept this current approach, we wouldn't want this as a `useEffect`. You'd want the request to start after you scanned your barcode. You'd have to create a fetch method for your `useFetch` hook that is executed once the bar code is scanned within `handleBarCodeScanned`. It's a more explicit way to do it instead of relying on a re-render to start the work for you.